src/bin/scheduler/scheduling/scheduler.rs

```diff
@@ -67,7 +69,18 @@ async fn run_sequential_plan_group_scheduler(
             _ = cancellation_token.cancelled() => { return }
         };
         for plan in plans.clone() {
-            let _ = spawn_blocking(move || run_plan(&plan).map_err(log_and_return_error)).await;
+            let plan_id = plan.id.clone();
+            match spawn_blocking(move || run_plan(&plan).map_err(log_and_return_error)).await {
+                Ok(Err(Terminate::Cancelled)) => {
+                    return;
+                }
+                Err(error) => {
```

Comment:
Does it make sense to add the explicit type in the match expression here? It might future proof us a little against API changes in `spawn_blocking`.

Reply:
Hmm, how would you do that? Not sure I understand what you mean.

Reply:
Ah, this only works in `python`, I guess? I was thinking along these lines:
```
Err(JoinError(e)) => {
```
But is probably not possible in Rust.